cpp/glfw.cpp

```diff
@@ -342,6 +342,13 @@ NAN_METHOD(createWindow) {
 		monitor = monitors[monitor_idx];
 	}
 	
+	#ifdef __APPLE__
+	glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 4);
```

Comment:
My understanding is that other systems will use their preferred GL version, but Apple requires it to be explicit.

Reply:
Doesn't if affect something like shader syntax or other stuff? It must remain compatible with what we have in 3d-core, with how shaders are written in three.js, etc. If there are some compatibility tweaks (on 3d-core side, for example) to be done, we must know and address them.

Also let's think about all the platforms, which versions should be used on linux and windows? I think with a decent GPU driver windows easily catches up to GL4. Linux might do as well, but I only have it on VM, and cant know for sure because I don't use it as much.

Reply:
I'm using GL 4.6 on Linux. Can't say for Windows because I only use that for games. MacOS seems pinned to 4.1.

Reply:
Well, I'd say "games" is the most convincing evidence for a good GL implementation, even though most of the windows games are DX, but still there are a lot of GL games.

I'm more concerned about the shaders here.